src/track/beats.h

```diff
@@ -82,8 +120,30 @@ class Beats : private std::enable_shared_from_this<Beats> {
             updateValue();
         }
 
+        /// Returns the frame duration between this beat and the next beat.
         mixxx::audio::FrameDiff_t beatLengthFrames() const;
 
+        /// Returns the number of beats per bar.
+        ///
+        /// Note: Currently always returns 4, as time signatures other than 4/4
+        /// are not supported yet.
+        int beatsPerBar() const;
+
+        /// Returns the current index of the beat in the bar.
+        int beatIndexInBar() const {
+            return m_beatOffset % beatsPerBar();
+        }
```

Comment:
I will have a look.

Reply:
don't waste your time on it for now, it was just a thought. I'm actually more concerned that the iterator operations may be inefficient (because operations such as `--` actually need $O(n)$ instead of $O(1)$.